### PR TITLE
Materialize references to *all* visible sets from inside pointers

### PR DESCRIPTION
**Pull request by msullivan**
There are still a couple of related-seeming issues, one of which is
pretty scary (test_edgeql_scope_source_rebind_01), so we're not done
yet, but this fixes a bunch.

### CODE REVIEW

File: edb/edgeql/compiler/viewgen.py [hunk 1]
```diff
@@ -1313,7 +1313,7 @@ def _compile_view_shapes_in_set(
             element = setgen.extend_path(
                 path_tip,
                 ptr,
-                unnest_fence=True,
+                unnest_fence=False,
```
Comment (elprans) [inline edb/edgeql/compiler/viewgen.py:1316]: Hm.  Curious why is this needed and why doesn't this trip `test_edgeql_ir_scope_tree_bad_03`?

Comment (msullivan) [reply edb/edgeql/compiler/viewgen.py:1316]: Having an `unnest_fence` there was causing spurious failures in cases like `test_edgeql_scope_ref_outer_03` and the new `test_edgeql_scope_ref_outer_07`, where we define an alias that refers to an outer scope and then use the alias in a semi-joiny way.

It doesn't trip `test_edgeql_ir_scope_tree_bad_03` because we use an unnest fence when normalizing ptr exprs

Comment (msullivan) [reply edb/edgeql/compiler/viewgen.py:1316]: Actually I should just drop the `unnest_fence` argument altogether if I'm doing this, too

Comment (elprans) [reply edb/edgeql/compiler/viewgen.py]: Oh, that's a nice cleanup!

**Review by elprans — APPROVED** [review]
